3rdparty/mbedtls/CMakeLists.txt

```diff
@@ -60,3 +61,36 @@ target_link_libraries(mbedcrypto INTERFACE
 target_link_libraries(mbedcrypto INTERFACE
 	${OE_LIBDIR}/openenclave/enclave/liboelibc.a
 	${OE_LIBDIR}/openenclave/enclave/liboecore.a)
+
+else()
+# On Windows, we compile the mbedcrypto library so we can use it for the bignum functions.
+
+if (CMAKE_GENERATOR MATCHES "Visual Studio")
+    set(MBEDTLS_LIBRARY_NAME "${CMAKE_BUILD_TYPE}/mbedcrypto.lib")
+else()
+    set(MBEDTLS_LIBRARY_NAME "mbedcrypto.lib")
+endif()
+
+ExternalProject_Add(mbedtls-wrap-host
+    SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/mbedtls
+
+    CMAKE_ARGS
+        -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+        -DENABLE_PROGRAMS=OFF
+        -DENABLE_TESTING=OFF
+        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+
+    INSTALL_COMMAND
+        COMMAND ${CMAKE_COMMAND} -E make_directory ${OE_LIBDIR}/openenclave/host/
+        COMMAND ${CMAKE_COMMAND} -E copy
+        <BINARY_DIR>/library/${MBEDTLS_LIBRARY_NAME} ${OE_LIBDIR}/openenclave/host/
+)
+install (FILES ${OE_LIBDIR}/openenclave/host/${MBEDTLS_LIBRARY_NAME} DESTINATION ${CMAKE_INSTALL_LIBDIR}/openenclave/host)
+
+ExternalProject_Get_Property(mbedtls-wrap-host BINARY_DIR)
+add_library(mbedcryptohost INTERFACE)
```

Comment:
Instead of creating an mbedtls library for host, does it make sense to just included the relevant mbedtls files in the CMakeLists.txt for host.
Currently it gives the impression that oe host requires multiple crypto libraries, whereas we just need a fraction of mbedtls capabilities.